apps/remote_control/test/lexical/remote_control/build/error_test.exs

```diff
@@ -9,28 +9,79 @@ defmodule Lexical.RemoteControl.Build.ErrorTest do
   def compile(source) do
     doc = Document.new("file:///unknown.ex", source, 0)
 
-    case Code.string_to_quoted(source) do
+    case Code.string_to_quoted(source, columns: true) do
```

Comment:
You've duplicated a lot of the code in `build.ex`. I think it's time to extract some of that code into its own modules; say `Build.Compile.File` and `Build.Compile.Project`, and you can use those modules here directly.

I also think you're going to need to add a tag like `@tag :with_diagnostics` that only runs in >= 1.15. 

This means you'll have to add the following to `test_helper.exs`
```elixir
ExUnit.configure(exclude: [with_diagnostics: function_exported?(Code, :with_diagnostics, 1)])
```

Reply:
Honestly, I don't quite understand why this tag `@tag :with_diagnostics` will runs in >= 1.15?

Reply:
because we define it to be true in the call to `ExUnit.configure` that I pasted in above.

Maybe a better condition would be:
```elixir
ExUnit.configure(exclude: [with_diagnostics: Version.match?(System.version(), "< 1.15.3")])
```

Reply:
Are you sure we need this? Did you check the latest test code? we don't call `with_diagnostics` directly now.